### ISSUE TITLE
chocolatey install ServiceBusExplorer doesnt work

### ISSUE THREAD

**Issue by vmachacek** [paolosalvatori/ServiceBusExplorer#137]
this is what it says:

```
Chocolatey v0.10.7
Installing the following packages:
ServiceBusExplorer
By installing you accept licenses for the packages.
ServiceBusExplorer not installed. The package was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""
 Source(s): "https://chocolatey.org/api/v2/"

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - ServiceBusExplorer - ServiceBusExplorer not installed. The package was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""
 Source(s): "https://chocolatey.org/api/v2/"
```

**Comment by paolosalvatori** [comment]
Thanks @vmachacek for the heads up. @SeanFeldman @tomkerkhove can you help here?

**Comment by tomkerkhove** [comment]
Looks like [no package is published](https://chocolatey.org/packages?q=ServiceBusExplorer).
Is the package published? Looking at #129, I think it's not?

**Comment by chhantyal** [comment]
FYI, there is lot of red text in package namespace https://chocolatey.org/packages/ServiceBusExplorer

**Comment by paolosalvatori** [comment]
What do you mean with red text? Do you mean that the latest version doesn't compile? Did you try to restore Nuget packages?

**Comment by tomkerkhove** [comment]
The package is not passing validation by chocolatey.

>This package was rejected on 1/30/2016. The reviewer chocolatey-ops has listed the following reason(s): 
Hello -
> - please add project url
> - please add a license url if you have one
> - Content icons folder - did you mean to include all of that?
> - The dependencies you linked do not exist. If they are dlls then please include them with the app - we do application level dependencies only - like git extensions depends on git.

Looks like it was submitted on 1/30/2016 but it wasn't allowed.
The chocolatey setup needs to be revised AFAIK, is this currently done automatically @SeanFeldman?

**Comment by paolosalvatori** [comment]
Hi @tomkerkhove @SeanFeldman, @Nate-Wilkins is working on this. See https://github.com/paolosalvatori/ServiceBusExplorer/pull/129

**Comment by Nate-Wilkins** [comment]
I've submitted the PR #140

**Comment by SeanFeldman** [comment]
Fixed via https://github.com/paolosalvatori/ServiceBusExplorer/commit/35f2c270d12e0a7a957df56d82be5d856dd587b1

**Comment by SeanFeldman** [comment]
@Nate-Wilkins sadly this is still not working and now we've got a broken master 😞
```
IF '%APPVEYOR_REPO_BRANCH%'=='master' IF '%APPVEYOR_PULL_REQUEST_NUMBER%'=='' choco push ServiceBusExplorer.%GitVersion_MajorMinorPatch%.nupkg -k="'%choco_token%''"
Chocolatey v0.10.7
Attempting to push ServiceBusExplorer.4.0.100.nupkg to https://push.chocolatey.org/
Failed to process request. 'The API key '*****'' is invalid.'. 
The remote server returned an error: (400) Bad Request..
Command exited with code 1
```

Could please address this? Thank you.

**Comment by SeanFeldman** [comment]
It looks like the token is passed as-is, meaning it will be encrypted token, not the actual value.

**Comment by Nate-Wilkins** [comment]
Looks like I wasn't using the wrong variable syntax for that line... sorry my appveyor knowledge is lacking.
Submitting a new PR now.

**Comment by SeanFeldman** [comment]
#141 raised to address the broken build. Thanks @Nate-Wilkins

**Comment by SeanFeldman** [comment]
Closed via https://github.com/paolosalvatori/ServiceBusExplorer/commit/e50fb03413470fc191be0bb4e2715bd30ecf0a76

**Comment by SeanFeldman** [comment]
![image](https://user-images.githubusercontent.com/1309622/28471923-fc53eed2-6dfb-11e7-94c5-2de6ae6d90b6.png)

**Comment by Nate-Wilkins** [comment]
🎉 🎉 🎉

**Comment by SeanFeldman** [comment]
Package is still not showing up, but I assume it's a matter of time.

**Comment by tomkerkhove** [comment]
It's still under evaluation but sure it will be fine!
![screenshot_2](https://user-images.githubusercontent.com/4345663/28474449-122325e6-6e49-11e7-9e38-f4e790d5a173.png)

**Comment by SeanFeldman** [comment]
👍

**Comment by Nate-Wilkins** [comment]
Yeah it takes them a bit to approve it but it's passed automated testing & validation so we should be in the clear.

If you want you can run `chocolatey install ServiceBusExplorer --confirm --pre --version 4.0.101` and `servicebusexplorer` to get it up and running 👍  works like a charm!

**Comment by paolosalvatori** [comment]
Cool, thanks guys!

**Comment by paolosalvatori** [comment]
Guys, I received the following comment on chocolatey. Could please fix the problem and clearly add my name and surname to the owners element? Thanks! Could you also copy the license.txt from the project in the package and add a verification.txt file? See comments below. Thanks! :)


Can you confirm that you're the software author through a comment here on chocolatey.org?
If not, please update the owners element to either your name or username.

Also, since binary files have been included inside the package a few changes are necessary.
1. The license of the software needs to be included inside the package as LICENSE.txt, this also needs to explicitly mention that you have the rights to distribute the binary files (unless you're also the software author, in which case distribution rights isn't needed).
2. A VERIFICATION.txt is also required to be added to the package, this should detail the download page where the binary file can be aquired, the direct download url to the binary files and the checksums of each binary file as well as the necessary steps to take to verify them.

Thanks,
Kim.

**Comment by tomkerkhove** [comment]
What a process to get this stuff in there. If nobody is up for it, I can do this later today/this week.

**Comment by paolosalvatori** [comment]
Thanks @tomkerkhove!

**Comment by tomkerkhove** [comment]
@paolosalvatori feel free to assign issue to me

**Comment by paolosalvatori** [comment]
Hi @tomkerkhove when you accept the invite as collaborator, I'll assign the task to you, thanks!

**Comment by tomkerkhove** [comment]
Re-opening as we need to pass validation first

**Comment by paolosalvatori** [comment]
Thanks @tomkerkhove

**Comment by Nate-Wilkins** [comment]
Yeah sorry, this is actually the first package I've uploaded from a source repository...

Thanks for sticking with this though guys - I appreciate it a lot!

**Comment by SeanFeldman** [comment]
@paolosalvatori there's an action you need to take on the [site itself](https://chocolatey.org/packages/ServiceBusExplorer).

> AdmiringWorm (reviewer) on 24 Jul 2017 11:49:59 +00:00:
> Can you confirm that you're the software author through a **comment here on chocolatey.org**?

Packaging `license.txt` shouldn't be an issue as it's checked out as part of the repo.

As for the `VERIFICATION.txt` 

>  this should detail the download page where the binary file can be aquired

https://github.com/paolosalvatori/ServiceBusExplorer/releases/latest will always point to the latest downloadable exe.

> the direct download url to the binary files and the checksums of each binary file as well as the necessary steps to take to verify them.

I'm not sure why do we need those files separately if we already have a ZIP file. For the checksums, we could generate those.

I suggest to contact the reviewer and question some of the scrutiny they've introduced in their process 😃

**Comment by paolosalvatori** [comment]
Hi @SeanFeldman read carefully: Can you confirm that you're the software author through a comment here on chocolatey.org? If not, please update the owners element to either your name or username.
Said that, I will comment on the site, even if I don't know where!

**Comment by tomkerkhove** [comment]
Since I know the release in AppVeyor I was going to point to the latest GH release and calculate the hash on that. This that is the best approach for this personally.

**Comment by SeanFeldman** [comment]
Nice. @tomkerkhove are you going to do it? Thanks.

**Comment by Nate-Wilkins** [comment]
@paolosalvatori would the checksum calculation be automated into appveyor then?

**Comment by SeanFeldman** [comment]
@Nate-Wilkins would need to be.

**Comment by tomkerkhove** [comment]
That's indeed how I was approaching it. I'm working on it @SeanFeldman but won't be ready today

**Comment by SeanFeldman** [comment]
Ok, a few things:
- We need to include LICENSE.txt within the nuget package
- We need to verify several hashes, not one ("checksums of each binary file as well as the necessary steps to take to verify them.")

Unfortunately, Chocolatey doco is... Has a bit of a chocolate color 💩 
@tomkerkhove and myself will work it out.

**Comment by SeanFeldman** [comment]
After looking into choco [documentation](https://chocolatey.org/docs) I think we're doing it wrong. 😱 

There's no need to create a nuget package and push it if the application is available as a zip file and we specify its URL 🙂 

Chocolatey has **helper scripts** to handle installation/modifications/uninstall scenarios. We should explore that first before going the custom nuget package route. Starting pointer: [`Install-ChocolateyZipPackage`](https://chocolatey.org/docs/helpers-install-chocolatey-zip-package) doco.

[VirtualBox choco package](https://github.com/chocolatey/chocolatey-coreteampackages/tree/master/automatic/virtualbox) does exactly that. It grabs virtualbox.exe from the web and deploys to chocolatey. @tomkerkhove lets first get that going.

@paolosalvatori could you please grant me access to chocolatey? Thanks.

**Comment by Nate-Wilkins** [comment]
The `Install-ChocolateyZipPackage` will still require a nuspec  file and a [`chocolateyInstall.ps1`](https://github.com/Nate-Wilkins/chocolatey-packages/blob/master/automatic/mysql-connector-java/tools/chocolateyInstall.ps1) script. The nupkg still gets pushed up but it just downloads it at a later point in time.

I'd guess that it's better to have the source files embedded into the nupkg but we can definitely switch it over.

**Comment by paolosalvatori** [comment]
I received the following comments from a Chocolatey reviewer:

flcdrg left the following comment(s):
License and Verification files need to be in the Chocolatey package, so that they can be viewed by users on the Chocolatey.org package page.
You can see examples for these files by running choco new demo
Also please remove the XML documentation files - (they're blowing up the web page code formatting scripts)

**Comment by tomkerkhove** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Service Bus Explorer - Chocolatey](https://trello.com/c/tajNJwGe/469-service-bus-explorer-chocolatey)

**Comment by tomkerkhove** [comment]
We'll look into this, I'm flying to NDC Sydney so will have enough time to kill..

They sure are not demanding...
> Also please remove the XML documentation files - (they're blowing up the web page code formatting scripts)

**Comment by paolosalvatori** [comment]
Take your time and have a good flight! I've no idea of what XML documentation we should remove. I'll eventually ask for more information. But I think they refer to *.xml files like Microsoft.ServiceBus.xml

**Comment by SeanFeldman** [comment]
@paolosalvatori that's exactly the one they are talking about.

**Comment by Nate-Wilkins** [comment]
@tomkerkhove any progress on this?
I can take over if you're too busy - I don't mind 👍

**Comment by tomkerkhove** [comment]
Was going to take a look at it on Thursday or this weekend. Up to you to decide!

**Comment by Nate-Wilkins** [comment]
Yeah I can jump on this! I'll post an update later tonight.

**Comment by tomkerkhove** [comment]
Ok great. Feel free to let me know what you've got. I was going to work on it this afternoon CEST, if you didn't finish it yet.

**Comment by Nate-Wilkins** [comment]
@tomkerkhove got something going let me know what you think! My powershell in appveyor is a bit... weird.

**Comment by tomkerkhove** [comment]
We've merged it in, let's hope this is sufficient for the Chocolatey team.
If not, I'll reopen this one. Feel free to keep us posted on the validation @paolosalvatori

**Comment by tomkerkhove** [comment]
Update - Passed automated checks and virus scanning, thanks @Nate-Wilkins. 

Still waiting on a moderator though - 

-  https://chocolatey.org/packages/ServiceBusExplorer
- 
https://chocolatey.org/faq#what-is-moderation

**Comment by SeanFeldman** [comment]
Hopefully this will be finished 🙂

**Comment by SeanFeldman** [comment]
Reopening this.

```
'ServiceBusExplorer' is Submitted.
Current status = Waiting for Maintainer to take corrective action

flcdrg left the following comment(s):

>Thanks for adding the license.txt. There doesn't seem to be any content in the verification.txt file.
>Please add that content. See choco new for examples.

If you are the software author, please confirm that via a comment here. Alternatively, if you are not the software author, put your own name in the owners field.
Make these changes and then re-submit the same version.

thanks,
David
```

We can't just patch, merge, and ask for a review, it needs to be the same version. I.e. no merging back into our `master` until choco tyrants are happy...  This is a nightmare if you ask me.

**Comment by tomkerkhove** [comment]
Working on a fix here https://github.com/tomkerkhove/ServiceBusExplorer/tree/patch/chocolatey

**Comment by SeanFeldman** [comment]
@tomkerkhove I've responded to Choco reviewers and now we need to ensure this one is addressed:

```
Thanks for confirming. Now just fill out tools\VERIFICATION.txt and we should be good to go.
```

**Comment by tomkerkhove** [comment]
@SeanFeldman PRing this in today, sorry about that. Forgot to move it to that folder

**Comment by tomkerkhove** [comment]
@SeanFeldman I'm not sure if I'm following here - Where did they comment this? When I check the [package page](https://chocolatey.org/packages/ServiceBusExplorer) I don't see that comment.

When I check the package the file is already in the tools folder:
![screenshot_1](https://user-images.githubusercontent.com/4345663/29744988-b4322ebc-8ab0-11e7-9f6a-869f07808249.png)

Content looks like:
![screenshot_2](https://user-images.githubusercontent.com/4345663/29744994-bce34c94-8ab0-11e7-880d-256f73f99afa.png)

Or are they referring to the templates they typically use? ie https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/automatic/7zip.install/legal/VERIFICATION.txt

**Comment by SeanFeldman** [comment]
Is this 103? That's on 102... Could we rename the nuget package to 102 and
push it? I know, not ideal.

**Comment by tomkerkhove** [comment]
I just took the output of latest AppVeyor build... This is getting confusing :)

**Comment by SeanFeldman** [comment]
Welcome to my reality 😃

**Comment by tomkerkhove** [comment]
Haha. I think the comment is outdated to be honest..  Was that a recent reply? 

Let's wait until they reject the current proposed package, maybe?

**Comment by SeanFeldman** [comment]
Yep, reply was today.

**Comment by paolosalvatori** [comment]
Hi @tomkerkhove @SeanFeldman @Nate-Wilkins I'm finally back from vacation. The latest message I received from Chocolatey is the following, but some of you has received it already:

'ServiceBusExplorer' is Submitted.
Current status = Waiting for Maintainer to take corrective action
flcdrg left the following comment(s):
Thanks for confirming. Now just fill out tools\VERIFICATION.txt and we should be good to go.
thanks,
David

**Comment by paolosalvatori** [comment]
Good morning guys @tomkerkhove @Nate-Wilkins @SeanFeldman I received the following comment in the weekend from Chocolatey:

flcdrg left the following comment(s):
Chocolatey.org already displays the checksums for embedded files.
Can you include in tools\verification.txt how to independently verify the files? If they come from https://github.com/paolosalvatori/ServiceBusExplorer/releases then mention that in the instructions.

**Comment by tomkerkhove** [comment]
I'll update the current approach to include the releases based on [this example](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/automatic/7zip.install/legal/VERIFICATION.txt).

Could you elaborate a bit on this please?
> Chocolatey.org already displays the checksums for embedded files.

What should be in the `verification.txt` then? Based on the above example I presume they just want a hash of the exe instead?

**Comment by SeanFeldman** [comment]
👍 @tomkerkhove 
Lets wrap this up. I'm having nightmares with this choco verification process.

**Comment by tomkerkhove** [comment]
Thanks for the review @SeanFeldman!

A new package has been pushed to Chocolatey with fixed value of v4.0.102.
Now we play the waiting game...

**Comment by paolosalvatori** [comment]
Thanks Tom, fingers crossed... :)

**Comment by SeanFeldman** [comment]
Sing Hallelujah!

```
ServiceBusExplorer' is Approved.
flcdrg left the following comment(s):

Status Change - Changed status of package from 'submitted' to 'approved'.

The package was approved by flcdrg on 9/9/2017.
```

@tomkerkhov can you raise another PR to remove hardcoded choco version? Once that is done, we can _finally_ close this issue.

**Comment by tomkerkhove** [comment]
I'm going to PR the cleanup this evening!

**Comment by paolosalvatori** [comment]
Thanks guys, you did it! Great collaboration :) Thanks a lot @tomkerkhove @SeanFeldman!

**Comment by SeanFeldman** [comment]
Closed via https://github.com/paolosalvatori/ServiceBusExplorer/commit/31bdd702e4a46f083d03842dd6d6483dc4e1803b

**Comment by SeanFeldman** [comment]
Will be released in milestone `4.0.103`. Right now staged in `develop`.

**Comment by rabrahamOpti** [comment]
I'm seeing this issue now:

![image](https://user-images.githubusercontent.com/28309959/46438901-a7b78980-c74e-11e8-9481-0c9faf36b39b.png)

**Comment by tomkerkhove** [comment]
Can you try using the correct casing and let us know if you are still having issues?

**Comment by rabrahamOpti** [comment]
Yup, that worked, thank you. FYI: I had copied/pasted the example line on chocolatey.org. That should be fixed:

![image](https://user-images.githubusercontent.com/28309959/46476107-58f10a80-c7b5-11e8-9781-078d28b70fe2.png)

**Comment by tomkerkhove** [comment]
That's odd. Unfortunately we don't control this and I suggest creating a dedicated issue on this repo so I can follow-up with choco team.

Are you up for that?

**Comment by rabrahamOpti** [comment]
Done, cheers:
https://github.com/paolosalvatori/ServiceBusExplorer/issues/264

**Comment by tomkerkhove** [comment]
Awesome, thanks!
